packages/model-viewer/src/three-components/gltf-instance/correlated-scene-graph.ts

```diff
@@ -122,35 +149,46 @@ export class CorrelatedSceneGraph {
     const defaultMaterial = {name: 'Default'} as Material;
     const defaultReference = {type: 'materials', index: -1} as GLTFReference;
 
+    // Gets the parser cache.
+    const parserCache =
+        new ParserCache(cloneThreeGLTF.parser as unknown as ParserInterface);
+
     for (let i = 0; i < originalThreeGLTF.scenes.length; i++) {
       this[$parallelTraverseThreeScene](
           originalThreeGLTF.scenes[i],
           cloneThreeGLTF.scenes[i],
           (object: ThreeSceneObject, cloneObject: ThreeSceneObject) => {
-            let elementReference =
+            const elementReference =
                 upstreamCorrelatedSceneGraph.threeObjectMap.get(object);
 
-            if (elementReference == null) {
+            const getMaterial = (mesh: Mesh) => {
+              return mesh.material as MeshStandardMaterial;
+            };
+            // Only sets up default material if the mesh references it.
+            if ((object as Mesh).isMesh &&
+                getMaterial(object as Mesh).uuid ===
+                    parserCache.defaultMaterialUUID) {
```

Comment:
Aside: it's nice to avoid putting top-level comments in the PR because github doesn't let you reply to them. I prefer to pick even a random line of code to attach them to instead, so that they can become a thread.

Reply:
> Found why there were extraneous empty materials and came up with a possible fix, it turns out we needed some more testing to narrow down the default-material cases to Meshes that are only using the default material. It seems maybe all scene objects where getting a default material element created.

I was just noticing this bug myself (and I introduced it) so thanks for fixing it! I suppose the UUID thing will work, but it's a tad complex. I wonder if the `isMesh` check is all you really need? Could you fix this by reverting all of this and just inserting that as a ternary where `elementReference` is defined?

Reply:
I guess that works too, actually, in that case why not just insert the default material once and be done, like so...

Reply:
hmm, isn't that just creating a default material all the time even if it's not needed again? Can we add a simple test while we're at it?

Reply:
Well originally it was generating one for every object not it just does it once, I had the check (to see if there is actually DefaultMaterial being used) to prevent this so it's only ever added in the legit cases. If that check is removed than, one will always be created so long as the gltf has a mesh in it, which is practically every gltf. But not sure if having a default representation there is a bad thing though.

Reply:
Fair, but I think it's still confusing to show a material that's unused. I think the even smaller change I suggested will fix both?

Reply:
New updates added, found another issue, which is that the Default material was being added by the original gltf correlation and then added again when cloned.

Reply:
Adding tests to catch these other issues such as multiple defaults, and extraneous materials, but this bit of code seems bug-prone, probably better to do this fix upstream by patching the incoming scene and gltf before this correlation step.

Reply:
Oh wow, I see how it's happening both in the original and in the cloning step now. I can't remember why I did that, but I agree it sounds like a terrible idea. Go for it.